components/merkledb/src/indexes/group.rs

```diff
@@ -119,10 +119,18 @@ where
     /// group in `suffixes` method and stores them in `Vec`. This operation can consume a large
     /// amount of RAM.
     ///
-    /// **Note:** creating any new indexes while this iterator is in scope may lead to panic.
-    pub fn keys_unchecked(&self) -> impl Iterator<Item = K::Owned> {
+    /// # Panics
+    ///
+    /// Panics if group prefix contains `id_in_group` component. Such group can be created via
+    /// `FromAccess` trait.
+    pub fn buffered_keys(&self) -> Vec<K::Owned> {
```

Comment:
Another thing: I think this method works incorrectly for `Prefixed` / `Migration` accesses. Unlike with `id_in_group`, we cannot afford to not support them, since the main use case for group enumeration is migration. Please add the corresponding tests.